src/main/java/org/madhuri/app/message/dao/MessageDAO.java

```diff
@@ -1,166 +1,107 @@
+		Transaction transaction = null;
+
+		try {
+			transaction = session.beginTransaction();
+			session.save(msg); // Save the new message entity
+			transaction.commit();
+		} catch (Exception e) {
+			if (transaction != null) {
+				transaction.rollback();
+			}
+			e.printStackTrace();
+		} finally {
+			session.close();
+		}
+
+		return msg;
+	}
+
+	public void deleteMessage(long id) {
+		Session session = HibernateUtil.getSession();
+		Transaction transaction = null;
+
+		try {
+			transaction = session.beginTransaction();
+			Message message = session.get(Message.class, id);
+			if (message != null) {
+				session.delete(message); // Delete the group entity
+				transaction.commit();
+			}
+		} catch (Exception e) {
+			if (transaction != null) {
+				transaction.rollback();
+			}
+			e.printStackTrace();
+		} finally {
+			session.close();
+		}
+	}
+
+	public Message getMessageById(long id) {
+		Session session = HibernateUtil.getSession();
+		Message message = session.get(Message.class, id);
+		session.close();
+		return message;
+	}
+
+	public Channel getChannelById(long channelId) {
+		Session session = HibernateUtil.getSession();
+		Channel channel = session.get(Channel.class, channelId);
+		session.close();
+		return channel;
 	}
-	
-	public List<Message> getMessagesByUsername(String username) {
-		List<Message> messages = new ArrayList<>();
-
-        try {
-            // 1. Get SQL connection
-            DbConnection dbConnection = new DbConnection();
-            Connection connection = dbConnection.getConnection();
-
-            // 2. Prepare the SELECT statement
-            String sql = "SELECT * FROM Message WHERE username=?";
-            PreparedStatement statement = connection.prepareStatement(sql);
-            statement.setString(1, username);
-
-            // 3. Execute the SELECT statement
-            ResultSet resultSet = statement.executeQuery();
-
-            // 4. Loop through the results and create message entities
-            while (resultSet.next()) {
-                Message message = createMessageFromResultSet(resultSet, username);
-                messages.add(message);
-            }
-            // 5. Close the resources
-            resultSet.close();
-            statement.close();
-            connection.close();
-
-        } catch (SQLException e) {
-            e.printStackTrace();
-        } catch (Exception e) {
-            e.printStackTrace();
-        }
-
-        return messages;
-    }
-	
+
+	public Message addMessageToChannel(long channelId, Message msg) {
+		try (Session session = HibernateUtil.getSession()) {
+			Transaction transaction = session.beginTransaction();
+
+			// Retrieve the channel by its ID
+			Channel channel = session.get(Channel.class, channelId);
+
+			if (channel != null) {
+				// Set the channelId in the message
+				msg.setChannelId(channelId);
+
+				// Add the message to the channel
+				channel.getMessages().add(msg);
+
```

Comment:
why do we have to add message to channel and update channel or saving a message in message table ?